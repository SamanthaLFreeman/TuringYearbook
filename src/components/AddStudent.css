form {
  display: flex;
  flex-direction: column;
  margin: auto;
  width: 30%;
}

input {
  border: none;
  margin: 0 0 10px 0;
  padding: 8px;
}

button {
  border: none;
  background-color: grey;
  padding: 8px;
}
