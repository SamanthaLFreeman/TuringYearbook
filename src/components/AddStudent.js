import React, { Component } from 'react';

class AddStudent extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      quote: "",
      superlative: ""
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({
      name: "",
      quote: "",
      superlative: ""
    })
  }

  render() {
    return (
      <form>
        <input 
        type="text" 
        placeholder="name" 
        name="name" 
        value={this.state.name} 
        onChange={this.handleChange}
></input>
        <input 
        type="text" 
        placeholder="quote" 
        name="quote" 
        value={this.state.quote} 
        onChange={this.handleChange}
></input>
        <input 
        type="text" 
        placeholder="superlative" 
        name="superlative"
        value={this.state.superlative}
        onChange={this.handleChange}
></input>
        <button
        onClick={(e) => this.handleSubmit(e)}>Submit</button>
      </form>
    )
  }
}

export default AddStudent;