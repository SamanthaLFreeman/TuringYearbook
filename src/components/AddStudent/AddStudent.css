form {
  display: flex;
  flex-direction: column;
  margin: auto;
  width: 30%;
}

input {
  border: none;
  margin: 0 0 10px 0;
  padding: 8px;
}

button {
  background-color: #035506;
  color: #fff;
  font-size: 14px;
  font-weight: bold;
  padding: 8px;
}
