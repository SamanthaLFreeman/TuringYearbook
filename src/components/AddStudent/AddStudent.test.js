import React from 'react';
import { shallow } from 'enzyme';
import AddStudent from './AddStudent';

describe('AddStudent', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<AddStudent addStudent={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })
})