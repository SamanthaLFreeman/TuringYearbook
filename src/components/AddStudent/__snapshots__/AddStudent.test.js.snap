// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStudent should match the snapshot with all data passed in correctly 1`] = `
<form>
  <input
    className="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="quote-input"
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <input
    className="superlative-input"
    name="superlative"
    onChange={[Function]}
    placeholder="Superlative"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
